File: 본문/java/Code_5_05_2.java
import java.util.*;

class Code_5_05_2 {
	public static void main(String[] args) {
		// 입력
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		int K = sc.nextInt();
		int[] X = new int[N + 1];
		int[] Y = new int[N + 1];
		for (int i = 1; i <= N; i++) {
			X[i] = sc.nextInt();
			Y[i] = sc.nextInt();
		}
		
		// 왼쪽 끝 x, 오른쪽 끝 x, 아래쪽 끝 y, 위쪽 끝 y를 전체 탐색(각각 번호가 i, j, k, l)
		long answer = (1L << 62); // 일단 나올 수 없는 값을 설정
		for (int i = 1; i <= N; i++) {
			for (int j = 1; j <= N; j++) {
				for (int k = 1; k <= N; k++) {
					for (int l = 1; l <= N; l++) {
						int cl = X[i]; // 왼쪽 끝의 x 좌표
						int cr = X[j]; // 오른쪽 끝의 x 좌표
						int dl = Y[k]; // 아래쪽 끝의 y 좌표
						int dr = Y[l]; // 위쪽 끝의 y 좌표
						if (check_numpoints(N, X, Y, cl, cr, dl, dr) >= K) {
							long area = (long)(cr - cl) * (dr - dl);
							answer = Math.min(answer, area);
						}
					}
				}
			}
		}
		
		// 답 출력
		System.out.println(answer);
	}
	static int check_numpoints(int N, int[] X, int[] Y, int lx, int rx, int ly, int ry) {
		int cnt = 0;
		for (int i = 1; i <= N; i++) {
			// 점(X[i], Y[i])가 직사각형에 포함되어 있는지 판정
			if (lx <= X[i] && X[i] <= rx && ly <= Y[i] && Y[i] <= ry) {
				cnt += 1;
			}
		}
		return cnt;
	}
}

File: 본문/python/Code_5_05_2.py
def check_numpoints(N, X, Y, lx, rx, ly, ry):
	cnt = 0
	for i in range(N):
		# 점(X[i], Y[i])가 직사각형에 포함되어 있는지 판정
		if lx <= X[i] and X[i] <= rx and ly <= Y[i] and Y[i] <= ry:
			cnt += 1
	return cnt

# 입력
N, K = map(int, input().split())
X = [ None ] * N
Y = [ None ] * N
for i in range(N):
	X[i], Y[i] = map(int, input().split())

# 왼쪽 끝 x, 오른쪽 끝 x, 아래쪽 끝 y, 위쪽 끝 y를 전체 탐색(각각 번호가 i, j, k, l)
answer = 10 ** 19  # 일단 나올 수 없는 값을 설정
for i in range(N):
	for j in range(N):
		for k in range(N):
			for l in range(N):
				# cl <= x <= cr, dl <= y <= dr인 직사각형
				# 직사각형이 만들어지려면, cl < cr, dl < dr이어야 하므로
				cl, cr, dl, dr = X[i], X[j], Y[k], Y[l]
				if cl < cr and dl < dr:
					if check_numpoints(N, X, Y, cl, cr, dl, dr) >= K:
						area = (cr - cl) * (dr - dl)
						answer = min(answer, area)

# 답 출력
print(answer)